op-service/clock/loop_test.go

```diff
@@ -0,0 +1,27 @@
+package clock
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLoopFn(t *testing.T) {
+	cl := NewDeterministicClock(time.Now())
+	calls := make(chan struct{}, 10)
+	testErr := errors.New("test close error")
+	loopFn := NewLoopFn(cl, func(ctx context.Context) {
+		calls <- struct{}{}
+	}, func() error {
+		close(calls)
+		return testErr
+	}, time.Second*10)
+	cl.AdvanceTime(time.Second * 15)
+	<-calls
+	cl.AdvanceTime(time.Second * 10)
+	<-calls
```

Comment:
I suggest adding
```suggestion
	<-calls
	select {
	case <-calls:
		t.Fatal("more calls than expected")
	default:
	}
```
to assert that exactly 2 calls have been made at this point, and not more.